h/static/scripts/directive/test/annotation-test.js

```diff
@@ -1460,137 +1353,42 @@ describe('annotation.js', function() {
         annotation.$update = sandbox.stub().returns(Promise.resolve());
         var controller = createDirective(annotation).controller;
         controller.edit();
-        controller.save();
-
-        // The controller currently removes the draft whenever an annotation
-        // update is committed on the server. This can happen either when saving
-        // locally or when an update is committed in another instance of H
-        // which is then pushed to the current instance.
-        annotation.updated = (new Date()).toISOString();
-        $scope.$digest();
-        assert.calledWith(fakeDrafts.remove, annotation);
+        return controller.save().then(function() {
+          assert.calledWith(fakeDrafts.remove, annotation);
+        });
       });
     });
 
-    describe('when the focused group changes', function() {
-      it('updates the current draft', function() {
+    describe('onGroupFocused()', function() {
```

Comment:
This was a deliberate change, and one I'd like to do to the rest of `annotation-test.js` as well. These "when the group changes" and "when it is a highlight" etc etc tests are a problem imo. It's not clear which part of the code each one of them is supposed to be testing, there are lots of tests that duplicate each other or that are no longer valid (but for obscure reasons still passing) or that test several disparate parts of the code at once etc as a result.

In the case of event listeners though, I think I'd be okay with using the name of the event rather than the name of the listener function, so `describe('when the GROUP_FOCUSED event is emitted')` or something.

Although just using the function name would be consistent with tests for other functions and methods.